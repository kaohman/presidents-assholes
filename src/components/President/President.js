import React from 'react';

const President = (props) => {
  const { number, president, birth_year, death_year, took_office, left_office, party } = props.president;
  return (
    <div className='president-card'>
      <h3>{president}</h3>
      <h4>{number}</h4>
      <h4>{party}</h4>
      <p>Birth: {birth_year}, Death: {death_year}</p>
      <p>Office: {took_office}-{left_office}</p>
    </div>
  )
}

export default President;