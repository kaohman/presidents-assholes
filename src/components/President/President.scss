.president-card {
  padding: 10px;
  margin: 20px;
  width: 280px;
  height: 300px;
  border: 1px solid black;
}