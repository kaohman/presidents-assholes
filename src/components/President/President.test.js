import President from './President';
import { shallow } from 'enzyme';
import React from 'react';

describe('President', () => {
  let wrapper;
  let presidentMock;

  beforeEach(() => {
    presidentMock = {
      number: 2,
      president: "John Adams",
      birth_year: 1735,
      death_year: 1826,
      took_office: "1797-03-04",
      left_office: "1801-03-04",
      party: "Federalist"
    };
    wrapper = shallow(
      <President
        president={presidentMock}
      />
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});