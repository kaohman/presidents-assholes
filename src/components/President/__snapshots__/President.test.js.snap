// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`President should match the snapshot 1`] = `
<div
  className="president-card"
>
  <h3>
    John Adams
  </h3>
  <h4>
    2
  </h4>
  <h4>
    Federalist
  </h4>
  <p>
    Birth: 
    1735
    , Death: 
    1826
  </p>
  <p>
    Office: 
    1797-03-04
    -
    1801-03-04
  </p>
</div>
`;
